client/src/components/revealComplete.js

```diff
@@ -1,41 +1,102 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import { observer } from 'mobx-react-lite';
+
 import styled from "styled-components";
 import Popup from "./popup";
 import { BoxContent, Header } from "../styles/BoxContent";
-import { Button } from "rimble-ui";
+import { Button, Loader } from "rimble-ui";
+import WaifuCard from "./waifuCard.js";
+
+import { RootStoreContext } from "../app/stores/root.store";
 
 const StyledRevealComplete = styled.div``;
 
+const RevealWaifusContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
 const ButtonContainer = styled.div`
   width: 100%;
   display: flex;
   justify-content: center;
   align-items: center;
 `;
 
-const RevealComplete = ({ show }) => {
+const getRevealedWaifus = async (WETStore) => {
+  const oldWaifuIds = WETStore.ownedItems.map(item => item.id);
+  await WETStore.syncOwnedItems();
+  const newWaifuIds = WETStore.ownedItems.map(item => item.id);
+  const newWaifus = WETStore.ownedItems;
+  return newWaifus.filter(waifu => !oldWaifuIds.includes(waifu.id));
+  return newWaifuIds.filter(newId => !oldWaifuIds.includes(newId));
+}
+
+const RevealComplete = observer(({ show }) => {
+  const rootStore = useContext(RootStoreContext);
+  const { WETStore } = rootStore;
+  const [revealedWaifuIds, setRevealedWaifuIds] = useState([]);
+  const [loading, setLoading] = useState(false);
+  
+  useEffect(
+    () => {
+      if (show) {
+        setLoading(true);
+        getRevealedWaifus(WETStore)
+        .then((value) => {
+          setRevealedWaifuIds(value);
+          setLoading(false);
+        }).catch((err) => {
+          setLoading(false);
+        });
+      }
+    },
+    [show, WETStore]
+  )
+
   return (
     <StyledRevealComplete>
       <Popup
         show={show}
         content={
           <BoxContent>
-            <Header>Waifu Revealed!!!</Header>
-            <ButtonContainer>
-              <Button.Outline
-                className="waifu-card-buttons"
-                onClick={() => {
-                  window.location.href = "/app";
-                }}
-              >
-                <span className="waifu-button-learnmore">View Waifus</span>
-              </Button.Outline>
-            </ButtonContainer>
+            {
+              loading 
+                ? 
+                  <Loader
+                    className="wallet-waifu-loader"
+                    color="#FBE55F"
+                    size="80px"
+                  />
+                :
+              <>
+                <Header>{revealedWaifuIds.length} Waifu{revealedWaifuIds.length > 1 && 's'} Revealed!!!</Header>
```

Comment:
@waifu-lover  it looks like with these new waifus, the header is becoming larger than the nav bar image.  
Can we try resolve this, maybe by changing the nav bar image to have `width: 100%` so that it fills the space?  
Or by setting a max width for the popup so it doesn't get too large?

This is what I'm referring to:
![image](https://user-images.githubusercontent.com/53957795/111927078-36354680-8b14-11eb-837f-72f95e1ee882.png)

Reply:
Yep, it was actually the `<button>` element adding some padding and a weird border. 

Looking much better but not quite exactly the same as the others.

![image](https://user-images.githubusercontent.com/80998645/111966479-13725480-8b4b-11eb-8a3d-c386447b89d8.png)